src/components/BluePrint/BluePrint.tsx

```diff
@@ -51,59 +233,389 @@ const BluePrint: React.FC = () => {
   );
 };
 
-function initBlueprint(
+// resize and offset image to fit nicely in the container
+function resizeOffsetBlueprintImage(
+  img$: fabric.Image,
+  containerWidth: number,
+  containerHeight: number
+) {
+  const imgWidth = img$.get('width') || 1;
+  const imgHeight = img$.get('height') || 1;
+
+  const widthToHeightRatio = imgWidth / imgHeight;
+
+  const isLandscape = widthToHeightRatio > 1;
+
+  let topImgOffset;
+
+  if (isLandscape) {
+    img$.scaleToWidth(containerWidth);
+
+    const heightToWidthRatio = imgHeight / imgWidth;
+    const scaledImgHeight = containerWidth * heightToWidthRatio;
+    topImgOffset = (containerHeight - scaledImgHeight) / 2;
+
+    img$.set({ top: topImgOffset });
+  } else {
+    console.warn(
+      'WARNING: Horizonal Initial image offseting is not implemented yet',
+    );
+    img$.scaleToHeight(containerHeight);
+  }
+
+  return img$;
+}
+
+const ZOOM_IN_MAX = 10;
+
+function getZoomFromMouseWheelEvent(
+  canvas$: fabric.Canvas,
+  e: fabric.IEvent<WheelEvent>
+) {
+  const delta = e.e.deltaY;
+  const zoom = canvas$.getZoom();
+  const zoomBy = ( 0.999 ** delta );
+  let newZoom = zoom * zoomBy;
+
+  if (newZoom > ZOOM_IN_MAX) newZoom = ZOOM_IN_MAX;
+
+  if (newZoom < 1) newZoom = 1;
+
+  return newZoom;
+}
+
+function handleZoomEvent(
+  canvas$: fabric.Canvas,
+  event$: fabric.IEvent<WheelEvent>
+) : void {
+
+  let newZoom = getZoomFromMouseWheelEvent(canvas$, event$)
+
+  const {offsetX, offsetY } = event$.e;
+
+  canvas$.zoomToPoint({ x: offsetX, y: offsetY }, newZoom);
+}
+
+// make sure canvas viewport stays within constraints
+function handleBorderConstraints(
```

Comment:
I think all this logic can be encapsulated inside hooks, as the file has grown a lot.